docs/DeveloperGuide.md

```diff
@@ -144,92 +139,88 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+**`Person`** component: 
+
+In Athena, contacts are represented by `Person` objects. `Person` objects have several properties such as email, address, etc. However, a `Person` can also be tagged with multiple `Tag`s.
+- `AddressBook` handles all direct matters concerning `Person` objects. It has a `TagManager` and `UniquePersonList`.  
+- `UniquePersonList` keeps track of all `Person` objects. Using `Person` class' `isSamePerson(Person)` method, it ensures that there are no duplicate contacts in Athena.
+- `TagManager` keeps track of which contacts contain which tags. It uses a hash map, with `Tag`s as keys and stores a set of `Person`s as its value.
+
+All manipulation of `Person` objects have to be done through `AddressBook`. `AddressBook` provides simple methods that can be used by higher-level components such as 
+- `void removePerson(Person)`
+- `void addPerson(Person)`
+- `boolean hasPerson(Person)`
+- and more...
+
+**`Tag`** component:
+
+`Tag`s are represented by a single alphanumeric string with no spaces. In v1.3, support for child-tagging was implemented. This allows directional relations to be established between `Tag`s. Certain commands also affect a `Tag` and its child-tags.
+- `TagTree` handles tag-to-tag relations 
+- `TagTreeImpl` extends from the abstract class `TagTree`. It uses a tree data structure to store directional tag-to-tag relations. The implementation of the tree is done with 
+a hash map, with `Tag`s used as keys, and its value being a set of all its child-tags.
+
+Any new links established between tags have to go through the `TagTree`. `TagTree` provides several simple methods such as 
+- `void addSubTagTo(Tag tag)`  
+- `boolean hasTag(Tag tag)`  
+- and more...
+
+**`Integration`** component:
+
+The `ContactTagIntegrationManager` class provides a few defined methods that affect both `Person`s and `Tag`s together. Methods include:
+- `void deleteTag(Tag)`
+- `void deleteTagRecursive(Tag)` - deletes a `Tag` and all its sub-tags
+- and more...
+
+The above methods listed methods are shown as examples to illustrate the difficulty of preserving consistency within the system.
+In particular, a method called `deleteTag` is implemented in both `TagTree` and `ContactTagIntegrationManager`.
+However, the method in `TagTree` only deletes the specific `Tag` in `TagTree`.
+The method in `ContactTagIntegrationManager` uses `TagTree`'s `deleteTag(Tag)` method and also removes the `Tag` from all `Person` objects that has the `Tag`.
+Below illustrates the sequence diagram for interactions between the `ContactTagIntegrationManager`, `AddressBook` and `TagTree` when `execute("delete -t t/cs2103")`.
+
+![delete-tag-demonstration](images/DeleteTagSequenceDiagram.png)
+
+As such, the `ContactTagIntegrationManager`'s job is to preserve consistency in the `Model` when a change is made to `Tag`s that will affect `Person`s stored.
+Thus, higher-level modules should use the methods in `ContactTagIntegrationManager` if possible.
```

Comment:
remove "if possible" to sound more confident in our implementation

Reply:
I changed it to "if available" to say that not every method related to tags and contacts is inside this class. Only those that might affect both at once. Some other methods will still need to be done with `AddressBook` and `TagTree`.